### PR TITLE
Mesh and Xenon platform deprecation

### PR DESCRIPTION
**Pull request by avtolstoy**
### Problem

Mesh deprecation.

### Steps to Test

N/A

### Example App

N/A

### References

- [CH47848]

---

### Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: hal/src/nRF52840/timer_hal.cpp [hunk 1]
```diff
@@ -15,14 +15,304 @@
...
+        __DSB();
+        __ISB();
+    }
+
+    nrf_rtc_event_clear(RTC_INSTANCE, NRF_RTC_EVENT_OVERFLOW);
+    nrf_rtc_event_enable(RTC_INSTANCE, RTC_EVTEN_OVRFLW_Msk);
+    nrf_rtc_int_enable(RTC_INSTANCE, NRF_RTC_INT_OVERFLOW_MASK);
+
+    nrf_rtc_event_clear(RTC_INSTANCE, NRF_RTC_EVENT_COMPARE_0);
+    nrf_rtc_event_disable(RTC_INSTANCE, NRF_RTC_EVENT_COMPARE_0);
```
Comment (sergeuz) [inline hal/src/nRF52840/timer_hal.cpp:227]: In OT they pass `RTC_EVTEN_COMPARE*_Msk` to `nrf_rtc_event_disable()` – does it matter which constant to use in our configuration? I can see two signatures of `nrf_rtc_event_disable()` in nrf_rtc.h.

File: system/src/system_cloud_internal_constants.c [hunk 2]
```diff
@@ -39,19 +39,6 @@ STATIC_ASSERT(backup_udp_public_server_key_size, sizeof(backup_udp_public_server
...
-const unsigned char backup_udp_public_server_address[] = {
-  0x01, 0x18, 0x24, 0x69, 0x64, 0x2e, 0x75, 0x64, 0x70, 0x2d, 0x6d, 0x65,
-  0x73, 0x68, 0x2e, 0x70, 0x61, 0x72, 0x74, 0x69, 0x63, 0x6c, 0x65, 0x2e,
-  0x69, 0x6f, 0x00
-};
-
-STATIC_ASSERT(backup_udp_public_server_address_size, sizeof(backup_udp_public_server_address)==27);
-
-# else // HAL_PLATFORM_MESH
-
```
Comment (sergeuz) [inline system/src/system_cloud_internal_constants.c:54]: :+1:

File: hal/src/nRF52840/timer_hal.cpp (hunk 1 above)
Comment (avtolstoy) [reply hal/src/nRF52840/timer_hal.cpp:227]: Good catch.
```
__STATIC_INLINE void nrf_rtc_event_disable(NRF_RTC_Type * p_rtc, uint32_t mask)
{
    p_rtc->EVTENCLR = mask;
}
```

Comment (avtolstoy) [reply hal/src/nRF52840/timer_hal.cpp:227]: Fixed.

File: hal/src/nRF52840/timer_hal.cpp [hunk 3]
```diff
@@ -15,14 +15,304 @@
...
+constexpr auto RTC_IRQ_PRIORITY = 6;
+constexpr auto US_PER_OVERFLOW = (512UL * 1000000ULL);  ///< Time that has passed between overflow events. On full RTC speed, it occurs every 512 s.
+#define RTC_IRQ_HANDLER RTC2_IRQHandler
+
+constexpr uint64_t divideAndCeil(uint64_t a, uint64_t b) {
+    return ((a + b - 1) / b);
+}
+
+constexpr uint64_t ticksToTime(uint64_t ticks) {
+    return divideAndCeil(ticks * (1000000 >> 6), 32768 >> 6);
```
[the remark this answers is not in the document]
Comment (avtolstoy) [reply hal/src/nRF52840/timer_hal.cpp]: Fixed.

File: communication/inc/communication_dynalib.h [hunk 4]
```diff
@@ -73,12 +73,12 @@ DYNALIB_FN(BASE_IDX2 + 1, communication, spark_protocol_command, int(ProtocolFac
 DYNALIB_FN(BASE_IDX2 + 2, communication, spark_protocol_time_request_pending, bool(ProtocolFacade*, void*))
 DYNALIB_FN(BASE_IDX2 + 3, communication, spark_protocol_time_last_synced, system_tick_t(ProtocolFacade*, time_t*, void*))
 
-#if HAL_PLATFORM_MESH
-DYNALIB_FN(BASE_IDX2 + 4, communication, spark_protocol_mesh_command, int(ProtocolFacade* protocol, MeshCommand::Enum cmd, uint32_t data, void* extraData, completion_handler_data* completion, void* reserved))
+#if HAL_PLATFORM_MESH_DEPRECATED
+DYNALIB_FN_PLACEHOLDER(BASE_IDX2 + 4, communication)
```
[the remark this answers is not in the document]
Comment (avtolstoy) [reply communication/inc/communication_dynalib.h]: Removed. It does shift things in the table but it's not a concern for Gen 3 as you said.

**Pull request merged**